src/main/java/com/exactpro/th2/readfile/Main.java

```diff
@@ -171,13 +173,21 @@ private static Unit publishError(MessageRouter<EventBatch> eventBatchRouter, Str
     }
 
     @NotNull
-    private static Unit publishMessages(MessageRouter<RawMessageBatch> rawMessageBatchRouter, StreamId streamId, List<Builder> builders) {
+    private static Unit publishMessages(
+            MessageRouter<GroupBatch> rawMessageBatchRouter,
+            StreamId streamId,
+            List<? extends RawMessage.Builder> builders,
+            Supplier<GroupBatch.Builder> batchSupplier
+    ) {
         try {
-            RawMessageBatch.Builder builder = RawMessageBatch.newBuilder();
-            for (RawMessage.Builder msg : builders) {
-                builder.addMessages(msg);
+            var batch = batchSupplier.get()
+                    .setSessionGroup(streamId.getSessionAlias());
+            for (var msg : builders) {
+                batch.addGroup(
+                        new MessageGroup(List.of(msg.build()))
+                );
             }
-            rawMessageBatchRouter.sendAll(builder.build());
+            rawMessageBatchRouter.sendAll(batch.build());
```

Comment:
Maybe use message batcher from common-utils library

Reply:
The batching is already performed on read-file-common-core side. We just need to send the resulting batch